.github/workflows/automated-release.yml

```diff
@@ -0,0 +1,99 @@
+name: Check for Version Updates and Create PR
+
+on:
+  schedule:
+    - cron: '0 0 * * *' # Trigger daily at midnight (adjust as needed)
+
+jobs:
+  check-and-create-pr:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
+
+      # Capture the current version tags for Pulsar and Advanced CLI
+      - name: Capture current version tags
+        id: capture_versions
+        run: |
+          current_pulsar_tag=$(grep -o "https://github.com/subspace/pulsar/releases/download/[^/]\+/[^/]\+" 'versioned_docs/version-latest/protocol/pulsar.mdx' | head -1 | awk -F'/' '{print $8}')
+          current_substrate_cli_tag=$(grep -o "https://github.com/subspace/subspace/releases/download/[^/]\+/[^/]\+" 'versioned_docs/version-latest/protocol/substrate-cli.mdx' | head -1 | awk -F'/' '{print $8}')
+          echo "current_pulsar_tag=${current_pulsar_tag}" >> $GITHUB_ENV
+          echo "current_substrate_cli_tag=${current_substrate_cli_tag}" >> $GITHUB_ENV
+
+      # Fetch the latest release tags for Pulsar and Advanced CLI
+      - name: Get latest Pulsar release
+        id: pulsar_release
+        run: |
+          response=$(curl -s -H "Authorization: Bearer ${{ secrets.GITHUB_TOKEN }}" \
+            "https://api.github.com/repos/subspace/pulsar/releases/latest")
+          echo "::set-output name=pulsar_tag::$(echo $response | jq -r '.tag_name')"
+
+      - name: Get latest Advanced CLI release
+        id: advanced_cli_release
+        run: |
+          response=$(curl -s -H "Authorization: Bearer ${{ secrets.GITHUB_TOKEN }}" \
+            "https://api.github.com/repos/subspace/subspace/releases/latest")
+          echo "::set-output name=advanced_cli_tag::$(echo $response | jq -r '.tag_name')"
+
+      # Compare the latest release tags with the current versions and update the documentation code
+      - name: Update documentation code
+        id: update-documentation-code
+        run: |
+          current_pulsar_tag=${{ env.current_pulsar_tag }}
+          current_substrate_cli_tag=${{ env.current_substrate_cli_tag }}
+          latest_pulsar_tag=${{ steps.pulsar_release.outputs.pulsar_tag }}
+          latest_advanced_cli_tag=${{ steps.advanced_cli_release.outputs.advanced_cli_tag }}
+
+          # Define arrays for whitelists and blacklists for each repository, each variable can hold an array such as ("first_word" "second_word")
+          pulsar_whitelist=("targetted-word")
+          pulsar_blacklist=("unwanted-word")
+          advanced_cli_whitelist=("gemini-3f")
+          advanced_cli_blacklist=("runtime" "snapshot")
+
+          # Check if the latest release should be updated based on whitelists and blacklists
```

Comment:
Consider a regular expression approach to do this. Maybe something for next time you're faced with a challenge that requires string matching.